kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProtocolFilterReconciler.java

```diff
@@ -0,0 +1,210 @@
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.ErrorStatusUpdateControl;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+
+import io.kroxylicious.kubernetes.api.common.Condition;
+import io.kroxylicious.kubernetes.api.common.ConditionBuilder;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilter;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilterBuilder;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Reconciles a {@link KafkaProtocolFilter} by checking whether the {@link Secret}s
+ * and {@link ConfigMap}s refered to in interpolated expressions actually exist, setting a
+ * {@link Condition.Type#ResolvedRefs} {@link Condition} accordingly.
+ */
+public class KafkaProtocolFilterReconciler implements
+        Reconciler<KafkaProtocolFilter> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaProtocolFilterReconciler.class);
+    private final Clock clock;
+    private final SecureConfigInterpolator secureConfigInterpolator;
+
+    KafkaProtocolFilterReconciler(Clock clock, SecureConfigInterpolator secureConfigInterpolator) {
+        this.clock = clock;
+        this.secureConfigInterpolator = secureConfigInterpolator;
+    }
+
+    @Override
+    public List<EventSource<?, KafkaProtocolFilter>> prepareEventSources(EventSourceContext<KafkaProtocolFilter> context) {
+        return List.of(
+                new InformerEventSource<>(templateResourceReferenceEventSourceConfig(context, Secret.class,
+                        interpolationResult -> interpolationResult.volumes().stream()
+                                .flatMap(volume -> Optional.ofNullable(volume.getSecret())
+                                        .map(SecretVolumeSource::getSecretName)
+                                        .stream())),
+                        context),
+                new InformerEventSource<>(templateResourceReferenceEventSourceConfig(context, ConfigMap.class,
+                        interpolationResult -> interpolationResult.volumes().stream()
+                                .flatMap(volume -> Optional.ofNullable(volume.getConfigMap())
+                                        .map(ConfigMapVolumeSource::getName)
+                                        .stream())),
+                        context));
+    }
+
+    /**
+     * Returns a new event source config for getting the resource dependencies of a given type present in a filter's {@code spec.configTemplate}.
```

Comment:
Nice. I definitely wouldn't have considered using an event sources to solve this problem.